### ISSUE TITLE
[12373] Custom Tab feature.

### ISSUE THREAD

**Issue by Aycorn** [DeskDirector/UserVoice#185]
This is currently available in the Admin Console but refers to legacy software:

![image](https://user-images.githubusercontent.com/30588784/47884828-31b25980-de97-11e8-9ffe-e73da7f5cd37.png)

Are there plans to have this feature on DD Tech?

**Comment by Lowell-Syncrety** [comment]
We use this feature a LOT so we'd not want to move away from the Staff HUD if this isn't in DD Tech.

**Comment by Andrew-Lahikainen** [comment]
@Lowell-Syncrety this is probably going to make it to tech

**Comment by Lowell-Syncrety** [comment]
Some things I use the custom tabs for in the Staff HUD are:

- General CWM tab; handy to view Companies, or whatever in CWM without having to leave Staff HUD.
- My CWM Calendar; great to manage my calendar in the CWM calendar as I have a ton of Activities right there; techs can also manage items there as the DD Tech calendar shows a list view but can't drag items to different times nor days.
- Service Board - I use this to get to different service boards but in my case just the Sales one, from which I create opportunities from sales tickets.
- My Time Sheets - We use this one all day; quick view of sequential time entries, adding new entries especially for non-service staff that use charge codes to track time.

**Comment by Andrew-Lahikainen** [comment]
I've renamed this issue to session tabs, because that's as far as we have gotten with our design. For main tabs I have created this issue: https://github.com/DeskDirector/UserVoice/issues/210

Session tabs have been implemented and are ready to be released. Release will likely be end of this week or beginning of next week.

**Comment by Lowell-Syncrety** [comment]
@Andrew-Lahikainen NICE!  That's a big step forward.

**Comment by Andrew-Lahikainen** [comment]
@Lowell-Syncrety are you using session tabs inside Staff HUD? Session tabs are the ones that appear when you open a ticket. Can I ask what you embed?

Also, I'll just point out here that most sites do not allow their pages to be iframed, which means we will have to pop open a new tab. I've tested with a couple of sites and the only one that has worked so far is ConnectWise. That's good news for you if you use a lot of ConnectWise embeds :)

**Comment by Lowell-Syncrety** [comment]
Oh, I thought these were the tabs across the top.  For session tabs, we definitely use the ConnectWise one.  I'd love to use the BizDox (CW Document) tab that I have there too but due to no integration it's a pain with credentials.

But in every ticket we always use the CW tab and that's where we do all service work.  So that way we have a nicely organized tab interface (Staff HUD) with CWM tabs in each session as needed, with the full functionality of CWM.

**Comment by Andrew-Lahikainen** [comment]
@Lowell-Syncrety it's immediate. It's probably not the UX you were expecting, but you will see a drop down button called "links" next to the 3 tabs on a ticket:

![image](https://user-images.githubusercontent.com/10571313/49322921-3db63780-f579-11e8-9e67-1a5b2a8e7100.png)

**Comment by Lowell-Syncrety** [comment]
Ah, I see that now.  So now just have to find where we go configure it to open the ticket in CWM, right in DD Tech as opposed to opening in a browser tab.  That way it keeps contained within DD Tech instead of into my already-overloaded browsers.

**Comment by Andrew-Lahikainen** [comment]
Are you talking about the open in psa button? I can treat that the same as these tabs and iframe it if possible. Will try get this into next release.

**Comment by Lowell-Syncrety** [comment]
Yes, that's it.  That's the tab we use 100% of the time in Staff HUD, where techs work all the time.

**Comment by Lowell-Syncrety** [comment]
So just checking, I can't remember which issue we were tracking this in... when we use the ConnectWise link in the ticket tab it opens the ticket just great, but when we switch to another ticket tab and do the same, the first ticket loses its ConnectWise session.

**Comment by Andrew-Lahikainen** [comment]
I'm not sure we are publicly tracking this improvement, but internally we will likely look into this next quarter (starts in 2 weeks).

Specifically the improvement is to make it so that tabs contain information on sub-states. We had to push this improvement back to next quarter because we have to focus on the new admin console implementation which contains a few new features.

It's actually quite a big refactor to implement this improvement in the way that I want because I want to tie a few other related things together.

Note that each time you navigate to a ticket that is on the connectwise link after this improvement, it will still need to reload connectwise. It will just save you two clicks, which can be a lot if you're switching often.

**Comment by Lowell-Syncrety** [comment]
Thanks for the update. Yes, it'll save 2 clicks but also a significant load time. The CW browser session takes a while to load, which isn't a big deal if you ever have to do it only once for a ticket.

**Comment by Andrew-Lahikainen** [comment]
I mean it will reload connectwise each time you navigate back to the dd-tech tab, so if you're switching between dd-tech tabs which are currently on the connectwise tab, it will reload connectwise. This means any information you enter inside connectwise might be lost as I don't think they save drafts or anything like that.

We can potentially make improvements around this, but it does introduce potential for performance problems. It also will significantly increase the internal complexity of tabs. I think it's best to not take on this complexity right now.

**Comment by Lowell-Syncrety** [comment]
Ah. We instruct techs to save as they add data in, because Staff HUD or their computer could crash anyway. But yeah, that's a difference between Staff HUD (which doesn't have to reload the CWM tab on ticket switch) and DD Tech.
